app/src/main/java/org/robojackets/apiary/ui/update/UpdateGate.kt

```diff
@@ -71,103 +17,127 @@ fun UpdateGate(
     content: @Composable () -> Unit,
 ) {
     content()
-    val updateState = rememberInAppUpdateState()
-    val scope = rememberCoroutineScope()
-    val result = updateState.appUpdateResult
-    val context = LocalContext.current
-    OnLifecycleEvent { owner, event ->
-        when (event) {
-            Lifecycle.Event.ON_RESUME -> {
-                if (result is AppUpdateResult.Available) {
-                    if (result.updateInfo.updateAvailability()
-                        == UpdateAvailability.DEVELOPER_TRIGGERED_UPDATE_IN_PROGRESS) {
-                        context.getActivity()
-                            ?.let { result.startImmediateUpdate(it, UPDATE_REQUEST_CODE) }
+    val inAppUpdateState = rememberInAppUpdateState(
+        highPrioritizeUpdates = 5,
+        mediumPrioritizeUpdates = 3,
+        promptIntervalHighPrioritizeUpdateInDays = 1,
+        promptIntervalMediumPrioritizeUpdateInDays = 4,
+        promptIntervalLowPrioritizeUpdateInDays = 8,
```

Comment:
Should these maybe be factored out to constants? I'm seeing the same numbers in several places. What happens if they get out of sync?

Reply:
It would cause inconsistent behavior on the update availability status in Settings vs. the prompt behavior. i'll figure something out